src/pint/utils.py

```diff
@@ -1667,3 +1673,83 @@ def remove_dummy_distance(c):
             "Do not know coordinate frame for %r: returning coordinates unchanged" % c
         )
         return c
+
+
+def info_string(prefix_string="# ", comment=None):
+    """
+    Returns an informative string about the current state of PINT.
+
+    Adds:
+    Creation date
+    PINT version
+    Username
+    Host
+    OS
+    plus a user-supplied comment (if present).
+
+    Username is given by the `gitpython` global configuration `user.name` (if available),
+    otherwise `getpass.getuser()`.
+
+    Parameters
+    ----------
+    prefix_string: str, default='# '
+        a string to be prefixed to the output (often to designate as a comment or similar)
+    comment: str, optional
+        a free-form comment string to be included if present
+    
+    Returns
+    -------
+    s : str
+        informative string
+
+
+    Notes
```

Comment:
Better to put examples in a standard Examples section: https://numpydoc.readthedocs.io/en/latest/format.html#examples
Also to give some sample code to produce such an example (I know that all you can supply is the comment and the comment character) like:
```
>>> info_string(prefix_string = "C ", comment="this is a \n two-line comment"
C Created: ...
```